src/test/java/plugins/GitPluginTest.java

```diff
@@ -209,6 +210,142 @@ public void update_submodules_recursively() throws IOException, InterruptedExcep
         job.startBuild().shouldSucceed();
     }
 
+    @Test
+    public void calculate_changelog() throws IOException, InterruptedException, SftpException, JSchException {
+        final URL changesUrl;
+        final String TEST_COMMIT_MESSAGE = "Second commit";
+        Build b;
+        GitRepo repo = buildGitRepo();
+
+        repo.branch("testBranch");
+        repo.commit(TEST_COMMIT_MESSAGE);
+        repo.transferToDockerContainer(host, port);
+
+        job.useScm(GitScm.class)
+                .url(repoUrl)
+                .credentials(USERNAME)
+                .calculateChangelog("origin", "testBranch");
+
+        job.save();
+
+        b = job.startBuild();
+        b.shouldSucceed();
+        changesUrl = b.url("changes");
+
+        assertThat(
+                b.getConsole(),
+                Matchers.containsRegexp("Using 'Changelog to branch' strategy.", Pattern.MULTILINE)
+        );
+
+        assertThat(
+                visit(changesUrl).getPageSource(),
+                Matchers.containsRegexp(TEST_COMMIT_MESSAGE, Pattern.MULTILINE)
+        );
+    }
+
+    @Test
+    public void clean_after_checkout() throws IOException, InterruptedException, SftpException, JSchException {
+        GitRepo repo = buildGitRepo();
+        repo.transferToDockerContainer(host, port);
+
+        //first config and build producing untrackedFile.txt
+
+        job.useScm(GitScm.class)
+                .url(repoUrl)
+                .credentials(USERNAME)
+                .cleanAfterCheckout();
+
+        job.addShellStep("touch untrackedFile.txt");
+        job.save();
+
+        job.startBuild().shouldSucceed();
+
+        // second config and build tests if file has been removed before checkout
+
+        job.configure();
+        job.removeFirstBuildStep();
+        job.addShellStep("ls && test ! -f untrackedFile.txt");
+        job.save();
+
+        job.startBuild().shouldSucceed();
+
+    }
+
+    @Test
+    public void clean_before_checkout() throws IOException, InterruptedException, SftpException, JSchException {
```

Comment:
The only thing I can think of spontaneously is that modified files could provoke a conflict when using clean_after_checkout... correct?